e2e/internal/e2e/profile.go

```diff
@@ -37,6 +39,8 @@ var (
 	UserNamespaceProfile = Profiles[userNamespaceProfile]
 	// RootUserNamespaceProfile is the execution profile for root and a user namespace.
 	RootUserNamespaceProfile = Profiles[rootUserNamespaceProfile]
+	// OverlayProfile is the execution profile for unprivileged overlayfs mount
+	OverlayProfile = Profiles[overlayProfile]
```

Comment:
If this is only for unprivileged overlayfs let's put that in the name.  How about UserOverlayProfile.